src/ImageSharp/Image/ImageProcessingExtensions.cs

```diff
@@ -41,13 +41,7 @@ public static Image<TColor> Apply<TColor>(this Image<TColor> source, IImageProce
         public static Image<TColor> Apply<TColor>(this Image<TColor> source, Rectangle sourceRectangle, IImageProcessor<TColor> processor)
             where TColor : struct, IPackedPixel, IEquatable<TColor>
         {
-            processor.Apply(source, sourceRectangle);
-
-            foreach (ImageFrame<TColor> sourceFrame in source.Frames)
-            {
-                processor.Apply(sourceFrame, sourceRectangle);
-            }
-
+            source.ApplyProcessor(processor, sourceRectangle);
```

Comment:
Surely no need for the extension method now if we are moving it to `ImageBase<TColor>`?